lib/rendition.js

```diff
@@ -60,6 +155,26 @@ class Rendition {
         return obj;
     }
 
+    inline() {
+        return Number.isInteger(this.instructions.inlineLimit)
+            && this.instructions.inlineLimit <= INLINE_LIMIT_MAX
+            && this.size() <= this.instructions.inlineLimit;
+    }
+
+    asDataUri() {
+        // should never happen as inlineLimit is limited, but extra safeguard against reading large files
+        if (this.size() > DATA_URI_LIMIT) {
+            throw new GenericError(`Rendition too large for data uri ${this.name}`);
```

Comment:
We have a `RenditionTooLarge` error. Would it be worthwhile to use it and have this as clarifying error message?

Reply:
No, as this would not be a _client_ error (which `RenditionTooLarge` is) but an error in our code, which should be visible as such.

With the current code it should be technically impossible to happen because of the other [check here](https://github.com/adobe/asset-compute-sdk/pull/12/files/6b89dea5624f7485801e9b75cd89404cb62e5710#diff-7758ccf59648f08e27a97a2510f1c2dfR160) as the comment says. It is just an extra safe guard in case the logic gets messed up in the future, since the file is read below, and we don't want to read large files by mistake.

Reply:
Validate for inline() ? What do you mean?

Reply:
the calling client does it [here](https://github.com/adobe/asset-compute-sdk/blob/6b89dea5624f7485801e9b75cd89404cb62e5710/lib/worker.js#L277) , so it seems like the logic can be encapsulated inside asDataUri()

Reply:
I think I like the way it is currently separated best. `asDataUri()` and `inline()` doing the most minimal job. Also the condition you point to is semantically clearer for a reader to understand... ? Or is it just my biased opinion?

Reply:
yes, better readability this way, just wonder if these(`asDataUri() and inline()`) functionally always go together.